File: app/src/main/java/com/savvasdalkitsis/gameframe/injector/infra/android/AndroidInjector.java
package com.savvasdalkitsis.gameframe.injector.infra.android;

import android.content.Context;
import android.net.wifi.WifiManager;

import com.savvasdalkitsis.gameframe.injector.ApplicationInjector;

public class AndroidInjector {

    public static WifiManager wifiManager() {
        return (WifiManager) ApplicationInjector.application().getSystemService(Context.WIFI_SERVICE);
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/injector/infra/android/AndroidInjector.kt
package com.savvasdalkitsis.gameframe.injector.infra.android

import android.content.Context
import android.net.wifi.WifiManager

import com.savvasdalkitsis.gameframe.injector.ApplicationInjector

object AndroidInjector {

    fun wifiManager() =
            ApplicationInjector.application().applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
}
